src/modules/simulation/gz_bridge/plugins/OpticalFlowSensor.cpp

```diff
@@ -0,0 +1,129 @@
+#include <gz/common/Console.hh>
+#include <gz/msgs/Utility.hh>
+#include <gz/sensors/Util.hh>
+
+#include "OpticalFlowSensor.hpp"
+#include "opticalflow.pb.h"
+
+using namespace custom;
+
+bool OpticalFlowSensor::Load(const sdf::Sensor &_sdf)
+{
+	auto type = gz::sensors::customType(_sdf);
+
+	if ("optical_flow" != type) {
+		gzerr << "Trying to load [optical_flow] sensor, but got type [" << type << "] instead." << std::endl;
+		return false;
+	}
+
+	gz::sensors::Sensor::Load(_sdf);
+
+	std::string output_topic = this->Topic();
+	_publisher = _node.Advertise<px4::msgs::OpticalFlow>(output_topic);
+	gzdbg << "Advertising optical flow data on: " << output_topic << std::endl;
+
+	std::string camera_topic = output_topic;
+	size_t last_segment = camera_topic.rfind("/optical_flow/optical_flow");
+
+	if (last_segment != std::string::npos) {
+		camera_topic = camera_topic.substr(0, last_segment) + "/flow_camera/image";
+	}
+
+	int image_width = 0;
+	int image_height = 0;
+	int update_rate = 0;
+	float hfov = 0;
+
+	auto sensorElem = _sdf.Element()->GetParent()->GetElement("sensor");
+
+	while (sensorElem) {
+		if (sensorElem->Get<std::string>("name") == "flow_camera") {
+			auto cameraElem = sensorElem->GetElement("camera");
+			update_rate = sensorElem->GetElement("update_rate")->Get<int>();
+			hfov = cameraElem->GetElement("horizontal_fov")->Get<double>();
+
+			auto imageElem = cameraElem->GetElement("image");
+			image_width = imageElem->GetElement("width")->Get<int>();
+			image_height = imageElem->GetElement("height")->Get<int>();
+			break;
+		}
+
+		sensorElem = sensorElem->GetNextElement("sensor");
+	}
+
+	gzdbg << "Camera parameters:" << std::endl
+	      << "  image_width: " << image_width << std::endl
+	      << "  image_height: " << image_height << std::endl
+	      << "  update_rate: " << update_rate << std::endl
+	      << "  hfov: " << hfov << std::endl;
+
+	gzdbg << "Subscribing to camera topic for flow: " << camera_topic << std::endl;
+
+	if (!_node.Subscribe(camera_topic, &OpticalFlowSensor::OnImage, this)) {
+		gzerr << "Failed to subscribe to camera topic: " << camera_topic << std::endl;
+		return false;
+	}
+
+	// Assume pinhole camera and 1:1 aspect ratio
+	float focal_length = (image_width / 2.0f) / tan(hfov / 2.0f);
```

Comment:
How similar is this to the optical flow sensors we are using? Anyway might be worth making this configurable.